native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/utils/CommonUtils.java

```diff
@@ -235,18 +234,29 @@ public static boolean checkForImportsInServices(List<Method> methodList, String
         return false;
     }
 
-    public static boolean checkForImportsInMessageMap(StubFile stubFile, String type) {
-        for (java.util.Map.Entry<String, Message> message : stubFile.getMessageMap().entrySet()) {
-            for (Field field: message.getValue().getFieldList()) {
-                if (field.getFieldType().equals(type)) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-
     public static boolean isType(String methodType, String type) {
         return methodType != null && methodType.equals(type);
     }
+
+    public static void addImports(StubFile stubFile, Set<String> ballerinaImports, Set<String> protobufImports) {
+        for (String protobufImport : stubFile.getImportList()) {
+            switch (protobufImport) {
```

Comment:
What about the `google.protobug.Struct`?

Reply:
The `google.protobuf.Struct` is a special case. The `ballerina/protobuf.types.struct` import is required when an input/ouput of `google.protobuf.Struct` type is present in a service function. However, if a message has a field of `google.protobuf.Struct` type, the `ballerina/protobuf.types.struct` import is not required since the corresponding Ballerina type is `map<anydata>`. eg:

**Proto 1 (import is needed):**
```
syntax = "proto3";

import "google/protobuf/struct.proto";

service StructHandler {
	rpc bidirectionalStreaming(stream google.protobuf.Struct) returns (stream google.protobuf.Struct) {}
}
```

**Proto 2 (Import is not needed):**
```
syntax = "proto3";

import "google/protobuf/struct.proto";

message Test {
    google.protobuf.Struct structDetails = 1;
}
```

Therefore, the import is added by iterating the service list[[1]](https://github.com/ballerina-platform/module-ballerina-grpc/blob/master/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java#L205).

Can we move this type to `ballerina/protobuf.types.struct` and import from it to make it consistent? 
@daneshk , @BuddhiWathsala WDYT

[1] https://github.com/ballerina-platform/module-ballerina-grpc/blob/master/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java#L205

Reply:
+1